lib/vm.js

```diff
@@ -400,6 +403,45 @@ const vmConstants = {
 
 ObjectFreeze(vmConstants);
 
+function stripTypeScriptTypes(code, options = {}) {
+  emitExperimentalWarning('vm.stripTypeScriptTypes');
+  const { mode = 'strip-only', sourceMap = false, filename = '' } = options;
+  validateObject(options, 'options');
+  validateString(mode, 'options.mode');
+  if (mode !== 'strip-only' && mode !== 'transform') {
+    throw new ERR_INVALID_ARG_VALUE('options.mode', mode, 'must be either ' +
+                                    '"strip-only" or "transform"');
+  }
+
+  if (mode === 'strip-only') {
+    if (sourceMap)
+      throw new ERR_INVALID_ARG_VALUE('options.sourceMap', mode, 'must be either ' +
+      'false or undefined when mode is "strip-only"');
+    if (filename)
+      throw new ERR_INVALID_ARG_VALUE('options.filename', mode, 'must be either ' +
+      'undefined or an empty string when mode is "strip-only"');
```

Comment:
consider using `validateOneOf` (though it won't have the suffix of `when mode is "strip-only"`)
```suggestion
    validateOneOf(sourceMap, 'options.sourceMap', [false, undefined]);
    validateOneOf(filename, 'options.filename', ['', undefined]);
```